File: CountryFlagAppwithoutrecycling/app/src/main/java/com/example/countryflagapp/MainActivity.java
package com.example.countryflagapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListView;

import java.sql.Array;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ListView countryListView;
    ArrayList<Country> countries = new ArrayList();

    CountryAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        populateTestData();

        countryListView = findViewById(R.id.countryListView);
        adapter = new CountryAdapter(this, countries);
        countryListView.setAdapter(adapter);
    }


    private void populateTestData() {
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));
        countries.add(new Country(101, "India", "in"));
        countries.add(new Country(101, "Malaysia", "my"));
        countries.add(new Country(101, "USA", "us"));
        countries.add(new Country(101, "Singapore", "sg"));
        countries.add(new Country(101, "Indonesia", "id"));
        countries.add(new Country(101, "France", "fr"));
        countries.add(new Country(101, "Indonesia", "it"));

    }
}

File: AppMemoryTest/app/src/main/java/com/example/appmemorytest/MainActivity.kt
package com.example.appmemorytest

import android.content.res.AssetManager
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.squareup.picasso.Picasso
import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.Date
import java.util.Timer
import java.util.TimerTask

class MainActivity : AppCompatActivity() {

    lateinit var timer:Timer
    lateinit var timer1:Timer

    var textArray = mutableListOf<String>()
    var bitmapArray = mutableListOf<Bitmap>()



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    


//        timer = Timer()
//        timer.scheduleAtFixedRate(object: TimerTask() {
//            override fun run() {
//                textArray.add(loadText())
//            }
//        }, Date(), 1000)
//
//        timer1 = Timer()
//        timer1.scheduleAtFixedRate(object: TimerTask() {
//            override fun run() {
//                bitmapArray.add(loadBitmap())
//            }
//        }, Date(), 1000)

    }

    fun loadBitmap(): Bitmap {
        val assets = getAssets()
        val fin = assets.open("ikigai.jpeg")
        val bitmap  = BitmapFactory.decodeStream(fin)
        return bitmap



    }


    fun loadText():String {
        val assets = getAssets()
        val fin = assets.open("test.txt")
        val inStr = InputStreamReader(fin)
        val bufferedStream = BufferedReader(inStr)
        var builder = StringBuilder()
        var line:String? = null
        line = bufferedStream.readLine()
        if(line != null) {
            builder.append(line)
        }

        while(line != null) {
            line = bufferedStream.readLine()
            if (line != null) {
                builder.append(line)
            }
        }

        return builder.toString()
    }

    override fun onTrimMemory(level: Int) {
        super.onTrimMemory(level)

        textArray.clear()
    }
}